analyzers/src/SonarAnalyzer.CSharp/Rules/LoopsAndLinq.cs

```diff
@@ -71,11 +71,11 @@ private static bool CanIfStatementBeMoved(IfStatementSyntax ifStatementSyntax)
         {
             return ifStatementSyntax.Else == null && (ConditionValidIsPattern() || ConditionValidInvocation());
 
-            bool ConditionValidIsPattern() => IsPatternExpressionSyntaxWrapper.IsInstance(ifStatementSyntax.Condition)
-                                         && !ifStatementSyntax.Condition.DescendantNodes()
-                                                                        .Any(d => d.IsAnyKind(SyntaxKindEx.VarPattern,
-                                                                                              SyntaxKindEx.SingleVariableDesignation,
-                                                                                              SyntaxKindEx.ParenthesizedVariableDesignation));
+            bool ConditionValidIsPattern() => (IsPatternExpressionSyntaxWrapper.IsInstance(ifStatementSyntax.Condition) || ifStatementSyntax.Condition.IsKind(SyntaxKind.IsExpression))
```

Comment:
Why did you add this `ifStatementSyntax.Condition.IsKind(SyntaxKind.IsExpression)` condition? Is it possible that  that part is true but `IsPatternExpressionSyntaxWrapper.IsInstance(ifStatementSyntax.Condition)` is false?

Reply:
It is indeed slightly weird, but the first one is C#10 and the second one is C#11 and _yes_, they are different.

If the right operand is `{Anything}Pattern`, like `VarPattern`, `ConstPattern`, etc it is an `IsPatternExpression` SyntaxKind.

If the right operand is a `PredefinedType`, like `string` it is an `IsExpression` SyntaxKind.

You can observe this by producing the AST for this trivial example (e.g. in sharplab):

```csharp 
var isExpression         = "hello" is string; // IsExpression
var isPatternExpression  = "hello" is {Length:42}; // IsPatternExpression
```

Reply:
You are correct I was just to tired and read on both sides IsPatternExpression :) Sorry for that. 
I that case I suggest to do the following check:
`ifStatementSyntax.Condition.IsAnyKind(SyntaxKind.IsExpression, SyntaxKindEx.IsPatternExpression)` and we can remove the
IsPatternExpressionSyntaxWrapper.IsInstance(ifStatementSyntax.Condition) check.